spiketoolkit/postprocessing/features.py

```diff
@@ -0,0 +1,143 @@
+"""
+Uses the functions in SpikeInterface/spikefeatures to compute
+unit template features
+"""
+
+import pandas as pd
+from spikefeatures import features
+from scipy.signal import resample
+from .postprocessing_tools import get_unit_templates, get_unit_max_channels
+
+
+def compute_unit_template_features(
+        sorting,
+        recording,
+        unit_ids=None,
+        feature_names=None,
+        save_property_or_features=False,
+        recovery_slope_window=0.7,
+        upsampling_factor=1,
+        invert_waveforms=False,
+        spread_threshold=None,
+        site_range=None,
+        site_spacing=None,
+        epoch_name=None,
+        channel_map=None,
+        return_dict=False,
+        verbose=False,
+
+):
+    """
+    Use SpikeInterface/spikefeatures to compute features for the unit
+    template. These consist of a set of 1D features:
+    - peak to valley (peak_to_valley), time between peak and valley
+    - halfwidth (halfwidth), width of peak at half its amplitude
+    - peak trough ratio (peak_trough_ratio), amplitude of peak over amplitude of trough
+    - repolarization slope (repolarization_slope), slope between trough and return to base
+    - recovery slope (recovery_slope), slope after peak towards baseline
+
+    And 2D features:
+
+    The metrics are computed on 'negative' waveforms, if templates are saved as
+    positive, pass keyword 'invert_waveforms'.
+
+    Recording should be bandpassfiltered  # TODO: make this a requirement?
+
+    Parameters:
+    -------
+    sorting: SortingExtractor
+        The sorting result to be evaluated.
+    recording: RecordingExtractor
+        The given recording extractor from which to extract amplitudes
+    unit_ids: list
+        List of unit ids to compute metric for. If not specified, all units are used
+    return_dict: bool
+        If True, will return dict of metrics (default False)
+    feature_names: list
+        list of feature names to be computed
+    upsampling_factor : int
+        factor with which to upsample the template resolution (default 1)
+
+    TODO:
+    save_property_or_features=False,
+    recovery_slope_window=0.7,
+    invert_waveforms=False,
+    spread_threshold=None,
+    site_range=None,
+    site_spacing=None,
+    epoch_name=None,
+    channel_map=None,
+    verbose=False,
+
+    Returns
+
+    -------
+    metrics : pandas.DataFrame
+        table containing all metrics:
+            rows: unit_ids
+            columns: metrics
+
+    OR
+    TODO
+    """
+
+    # ------------------- SETUP ------------------------------
+    if unit_ids is None:
+        unit_ids = sorting.get_unit_ids()
+
```

Comment:
Same thing for `channel_ids` here and then loop only through them

Reply:
ok, but should we call the function using `unit_ids` OR `channel_ids`?

Reply:
How does the `channel_ids` input argument look like? Can we have multiple channels per unit?

Reply:
I think we can do either:
pass `unit_ids` and `channel_ids`; `channel_ids` is a `dict`  or a `list with lists` with id's per channel

or 

Only allow 1 channel per unit; `channel_ids` is a list of length `unit_ids`

In either case, if  `channel_ids` are not given it defaults to the detected max_channel

Reply:
No, it's a single list. Otherwise in `get_unit_waveforms` we also have a `max_channels_per_waveforms` arg that selects the max channels separately for each unit.

Don't worry about that for now, I can add it later (and I think it coud be useful)

Reply:
or to ask this differently; do you see use for people passing multiple channels per unit?

Reply:
allright

Reply:
I do, but I think it would be easier to select the best channels internally (already implemented [here](https://github.com/SpikeInterface/spiketoolkit/blob/master/spiketoolkit/postprocessing/postprocessing_tools.py#L1657-L1676)).

So in my opinion one could:
- pass `channel_ids`, i.e. a single list of channels used for all unit_ids
- do not pass `channel_ids` and pass `max_channels_per_features`, which selects the `max_channels_per_features` best channels for that unit 
- I think that by default, `max_channel_per_features` should be 1, as most people probably extract feats from the max channel

What do you think?

Reply:
If `channel_ids` is used, it overrides `max_channel_per_waveforms`